src/main/java/dbaccessor/message/MessageAccessor.java

```diff
@@ -42,9 +51,21 @@ public void updateReceivedTs(Message message) {
     /**
      * Completes all DB insertions for the CreateMessage API in a single transaction.
      */
-    public boolean createMessageInTransaction(Message message, Chat chat) {
-        return spannerOperations.performReadWriteTransaction(
+    public boolean createMessageInTransaction(Message message, Chat chat, Optional<Attachment> optionalAttachment) {
```

Comment:
Using `Optional` as a parameter in public methods is an anti-pattern. You could instead create an overload of this method

Reply:
Done